### DISCUSSION TITLE
UnstyledButton TS can not be cast

### DISCUSSION THREAD

**Discussion by mazipan** [mantinedev/mantine#1115] [Q&A] [answered]
In the doc, we can cast the `Button` in TS like this:

```ts
import { Button } from '@mantine/core';
import { Link } from 'react-router-dom';

function Demo() {
  return (
    <>
      <Button<'a'> component="a" href="#" />
      <Button<typeof Link> component={Link} to="/hello" />
    </>
  );
}
```

But seems not working with `UnstyledButton`:

```ts
import { UnstyledButton } from '@mantine/core';
import { Link } from 'react-router-dom';

function Demo() {
  return (
    <>
      <UnstyledButton<'a'> component="a" href="#" />
      <UnstyledButton<typeof Link> component={Link} to="/hello" />
    </>
  );
}
```

I got this error on my text editor:

![Screen Shot 2022-03-28 at 10 16 28](https://user-images.githubusercontent.com/7221389/160320596-309dd1f7-9573-4b0a-bc0f-e456c175b1bf.png)

In case you need a minimal reproduction, please see https://codesandbox.io/s/mantine-unstyled-18ss6i?file=/src/App.tsx

Not sure if this is a real issue or just me wrong setup my TS.
Any help will be very appreciated.

**Comment by rtivital** [comment, marked as answer]
Currently, UnstyledButon is not a polymorphic component and does not support this feature.
